src/github.com/couchbase/sync_gateway/base/bucket_gocb.go

```diff
@@ -476,13 +496,45 @@ func isRecoverableGoCBError(err error) bool {
 	if err == nil {
 		return false
 	}
+
+	goCBErr := GoCBErrorType(err)
+
+	_, ok := recoverableGoCBErrors[goCBErr]
+
+	return ok
+}
+
+// GoCB error types - workaround until gocb has public error type lookup support
```

Comment:
I was waiting on that PR that someone had already submitted, but I agree it might be time to file our own ticket.